pkg/jobservice/service.go

```diff
@@ -0,0 +1,297 @@
+package jobservice
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/cedana/cedana/pkg/api/containerd"
+	"github.com/cedana/cedana/pkg/api/runc"
+	"github.com/cedana/cedana/pkg/api/services/task"
+	"github.com/cedana/cedana/pkg/jobservice/jobdb"
+
+	"github.com/rs/zerolog/log"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+//go:embed schema.sql
+var schemaSetup string
+
+func New() (*JobService, error) {
+	// sqlite queue
+	db, err := sql.Open("sqlite3", ":memory:?_journal=WAL&_timeout=5000&_fk=true")
+	if err != nil {
+		return nil, err
+	}
+	db.SetMaxOpenConns(1)
+	db.SetMaxIdleConns(1)
+
+	if _, err := db.ExecContext(context.Background(), schemaSetup); err != nil {
+		return nil, err
+	}
+
+	config, err := detectConfig()
+	if err != nil {
+		// couldn't figure out runtime
```

Comment:
worth creating an error here that explicitly says so because detectConfig just returns nil